File: 백준/Bronze/2875. 대회 or 인턴/대회 or 인턴.py
import sys
input=lambda:sys.stdin.readline().rstrip()
n,m,k=map(int,input().split())
teams=0
while True:
    n-=2
    m-=1
    if n<0 or m<0 or n+m <k:
        break
    teams+=1
print(teams)

File: 백준/Bronze/2875. 대회 or 인턴/대회 or 인턴.cc
#include <iostream>
using namespace std;
int main(void) {
    int n, m, k, teams = 0;
    cin >> n >> m >> k;
    while (true) {
        n -= 2;
        m -= 1;
        if (n < 0 || m < 0 || n + m < k) break;
        teams += 1;
    }
    cout << teams;
    return 0;
}